pkg/core/candidate/broker.go

```diff
@@ -195,3 +199,11 @@ func (b *Broker) clearEligibleHashes() {
 	}
 	b.lock.Unlock()
 }
+
+func (b *Broker) clearRepublishedHashes() {
+	b.lock.Lock()
```

Comment:
according to our guidelines, unexported functions should delegate locking to the caller, so to prevent deadlocks and excessive locking